src/aggregate_with_randomness.h

```diff
@@ -0,0 +1,149 @@
+#pragma once
+#include "addon.h"
+
+namespace {
+Napi::Value AggregateWithRandomness(const Napi::CallbackInfo &info) {
+    BLST_TS_FUNCTION_PREAMBLE(info, env, module)
+    Napi::Value array_value = info[0];
+    if (!array_value.IsArray()) {
+        Napi::TypeError::New(
+            env, "Must pass an array to aggregateWithRandomness")
+            .ThrowAsJavaScriptException();
+        return env.Undefined();
+    }
+    Napi::Array array = array_value.As<Napi::Array>();
+    uint32_t array_length = array.Length();
+    if (array_length == 0) {
+        Napi::TypeError::New(
+            env, "Empty array passed to aggregateWithRandomness")
+            .ThrowAsJavaScriptException();
+        return env.Undefined();
+    }
+
+    blst::P1 aggregate_key{};
+    blst::P2 aggregate_sig{};
+
+    for (uint32_t i = 0; i < array_length; i++) {
+        blst::byte randomness[BLST_TS_RANDOM_BYTES_LENGTH];
+        module->GetRandomBytes(randomness, BLST_TS_RANDOM_BYTES_LENGTH);
+        Napi::Value set_value = array[i];
+        if (!set_value.IsObject()) {
+            Napi::TypeError::New(
+                env, "Must pass an array of objects to aggregateWithRandomness")
+                .ThrowAsJavaScriptException();
+            return env.Undefined();
+        }
+        Napi::Object set = set_value.As<Napi::Object>();
+
+        try {
+            Napi::Value val = set.Get("publicKey");
+            if (val.IsTypedArray()) {
+                Napi::Uint8Array typed_array = val.As<Napi::Uint8Array>();
+                if (std::optional<std::string> err_msg =
+                        blst_ts::is_valid_length(
+                            typed_array.ByteLength(),
+                            blst_ts::public_key_length_compressed,
+                            blst_ts::public_key_length_uncompressed)) {
+                    Napi::TypeError::New(
+                        env, "BLST_ERROR: PublicKeyArg"s + *err_msg)
+                        .ThrowAsJavaScriptException();
+                    return env.Undefined();
+                } else if (blst_ts::is_zero_bytes(
+                               typed_array.Data(),
+                               0,
+                               typed_array.ByteLength())) {
+                    Napi::TypeError::New(
+                        env, "BLST_ERROR: PublicKeyArg must not be zero key")
+                        .ThrowAsJavaScriptException();
+                    return env.Undefined();
+                } else {
+                    blst::P1 key{typed_array.Data(), typed_array.ByteLength()};
```

Comment:
do we need an option to keyVerify and sigVerify?

Reply:
good idea.  Adding option to `validateSerialized` values only because we will be passing already deserialized and validated keys and it will slow things down a lot if we check those.  Will make a note in the docstring about it.

Reply:
fiddlesticks!!!! that was the performance difference.... Once validation runs the times are within 3-5% of the JS version.... I would not have guessed it would be so little difference.  I am going to make this async to see if shifting the work to libuv will help node performance.... TBD

Reply:
Yeah, at least taking the work off-thread might be helpful